brainsets/utils/split.py

```diff
@@ -168,3 +169,190 @@ def generate_train_valid_test_splits(epoch_dict, grid):
+    Subdivides intervals into fixed-length epochs using Interval.arange().
+
+    If some intervals are shorter than the duration, keep them as they are.
+    If an interval is not a perfect multiple of the duration, the last chunk will be shorter.
+
+    Args:
+        intervals: The original intervals to chop.
+        duration: The duration of each chopped interval in seconds.
+        check_no_overlap: If True, verify the resulting intervals don't overlap.
+
+    Returns:
+        Interval: A new Interval object containing the chopped segments.
+                  Metadata from the original intervals is preserved and repeated for each segment.
+
+    Raises:
+        ValueError: If check_no_overlap is True and intervals overlap.
+    """
+    if len(intervals) == 0:
+        return Interval(start=np.array([]), end=np.array([]))
+
+    chopped_intervals = []
+    original_indices = []
+
+    for i, (start, end) in enumerate(zip(intervals.start, intervals.end)):
+        if end - start <= duration:
+            chopped = Interval(start=start, end=end)
+        else:
+            chopped = Interval.arange(start, end, step=duration, include_end=True)
+
+        chopped_intervals.append(chopped)
+        original_indices.extend([i] * len(chopped))
+
+    all_starts = np.concatenate([c.start for c in chopped_intervals])
+    all_ends = np.concatenate([c.end for c in chopped_intervals])
+
+    kwargs = {}
+    if hasattr(intervals, "keys"):
+        for key in intervals.keys():
+            if key in ["start", "end"]:
+                continue
+            val = getattr(intervals, key)
+            if isinstance(val, (np.ndarray, list)) and len(val) == len(intervals):
+                if isinstance(val, np.ndarray):
+                    kwargs[key] = val[original_indices]
+                else:
+                    kwargs[key] = [val[i] for i in original_indices]
+
+    result = Interval(start=all_starts, end=all_ends, **kwargs)
+
+    if check_no_overlap:
+        if not result.is_disjoint():
+            raise ValueError("Intervals overlap after chopping")
+
+    return result
+
+
+def filter_intervals(
+    intervals: Interval, exclude_values: list, stratify_by: str = "id"
+) -> Interval:
+    """
+    Filters out intervals with specified values in a given attribute.
+
+    Args:
+        intervals: The intervals to filter.
+        exclude_values: A list of values to exclude.
+        stratify_by: The attribute name to use for filtering. Defaults to "id".
+
+    Returns:
+        Interval: A new Interval object with excluded values removed.
+
+    Raises:
+        ValueError: If the intervals don't have the specified attribute.
+    """
+    if not hasattr(intervals, stratify_by):
+        raise ValueError(
+            f"Intervals must have a '{stratify_by}' attribute for filtering."
+        )
+
+    attr_values = getattr(intervals, stratify_by)
+    mask = ~np.isin(attr_values, exclude_values)
+    return intervals.select_by_mask(mask)
```

Comment:
This whole function seems to be replaceable with a simple one-liner.

```python
x.select_by_mask(np.isin(x.id, exclude_values, invert=True))
```

You sure you want to have this as a function? 

If so, we should follow API signature of another library that has this (idk if numpy/pandas have this). We should rename parameter `stratify_by` to something more reasonable.